Troubleshooting findo3de (#1689)
o3de/o3de.org: content/docs/user-guide/build/troubleshooting.md

After cleaning the cache, the correct compiler should be detected during the CMake configuration stage.

## CMake fails to find `Findo3de.cmake` 

**Issue:** The CMake tool reports it cannot find a package configuration file provided by "o3de". This produces a warning similar to:

```cmd
CMake Error at CMakeLists.txt:10 (find_package): 

By not providing "Findo3de.cmake" in CMAKE_MODULE_PATH this project has 
asked CMake to find a package configuration file provided by "o3de", but 
CMake did not find one. 

Could not find a package configuration file provided by "o3de" with any of 
the following names: 

o3deConfig.cmake 

o3de-config.cmake

```

**Remedy:** This issue is usually caused by a misconfigured `o3de_manifest.json` file that has an entry for a version of O3DE that no longer exists or has missing or corrupt files.  This can occur if you installed O3DE and then deleted the engine folder instead of uninstalling it using "Add or Remove Programs".  Do the following:

> 1. Close `o3de.exe` if it is open.

Reviewer: Clarify which application this is, for people who might not know that `o3de.exe` is Project Manager.
```suggestion
1. Close Project Manager (`o3de.exe`) if it is open.
```

1. Open the `<user>/.o3de/o3de_manifest.json` file in a text editor.
1. Remove the invalid entries from "engines" and "engines_path"
1. Make sure that the final entries in "engines" and "engines_path" do not end in a comma.
1. Save and close the file.
1. Open `o3de.exe`

> 1. Re-add your project if you don't see it in the `Projects` page.

Reviewer 2: Well the first step we should do is to request the use lookup the engine their project is using by either examining the "<project-root>/project.json" for the project.

Do we show the "engine" field for the project in the Project Manager? That would be another way for the user to determine the name of the engine their project is using.

Also we don't have a definition of term "invalid" for the "engines" field.
If I was a user reading this section I would want to know what makes an "invalid" `engines` and `engines_path` entry.
Being explicit and stating removing entries within non-existent paths would be clearer

Author: Good suggestions, will address.

 Also I added an issue to track further investigation based on user feedback https://github.com/o3de/o3de/issues/9931

We do show the engine field in Project Manager, but it only shows the current installed engine.  If the invalid engine entry is for an uninstalled/deleted version of the engine, you wouldn't have a version of Project Manager to run that could show it.

> 1. Open `o3de.exe`

Reviewer: ```suggestion
1. Restart Project Manager (`o3de.exe`).
```

> When `o3de.exe` opens it should automatically re-register the O3DE engine and update `o3de_manifest.json` with the correct information. 

Reviewer: ```suggestion
When Project Manager opens it should automatically re-register the O3DE engine and update `o3de_manifest.json` with the correct information. 
```
